Modules/Core/GPUCommon/src/itkGPUKernelManager.cxx

```diff
@@ -324,8 +324,79 @@ GPUKernelManager::SetKernelArg(int kernelIdx, cl_uint argIdx, size_t argSize, co
   return true;
 }
 
+
+bool
+GPUKernelManager::SetKernelArgWithChar(int kernelIdx, cl_uint argIdx, char argVal)
+{
+  return this->SetKernelArg(kernelIdx, argIdx, sizeof(char), &argVal);
```

Comment:
Is it really necessary to pass the size of `argVal`  each time `this->SetKernelArg` is called (10 times)? It looks to me like a template function could be of help here.

Reply:
For C++ it is totally redundant, I though that commit message said it eases access from Python.

Reply:
Sorry I don't have time for a complete review of all these (1000+) lines of code, so I just focused on some of the C++ work in there.  These ten `sizeof()`'s appear redundant and avoidable. As far as I can see, these are part of the C++ implementation, not the Python interface.

Reply:
Oh, now I see what you meant.

Reply:
@N-Dekker addressed in 7ba119454b78dddd9d9f5d549207ba4030e09753